### PR TITLE
feat(app): track run method analytics [RAUT-221]

### PR DESCRIPTION
**Pull request by ewagoner**
# Overview

Wire up analytics to track run method [RAUT-221]

# Changelog

* Add sourceLocation property to proceedToRun tracker
* Add tracking to HistoricalProtocolRun
* Add tracker to RunRecordDetail
* Add tracker to ProtocolDetail
* Add tracker to ProtocolLanding
* Added tests for new tracking calls

# Review requests

N/A

# Risk assessment

N/A

[RAUT-221]: https://opentrons.atlassian.net/browse/RAUT-221?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by jerader — APPROVED** [review]
LGTM good job! 🚀

File: app/src/organisms/ProtocolDetails/index.tsx [hunk 1]
```diff
@@ -367,6 +369,14 @@ export function ProtocolDetails(
     ),
   }
 
+  const handleRunProtocolButtonClick = (): void => {
+    trackEvent({
+      name: 'proceedToRun',
+      properties: { sourceLocation: 'ProtocolDetail' },
```
Comment (koji) [inline app/src/organisms/ProtocolDetails/index.tsx]:
```suggestion
      properties: { sourceLocation: 'ProtocolDetails' },
```

File: app/src/organisms/ProtocolsLanding/__tests__/ProtocolOverflowMenu.test.tsx [hunk 2]
```diff
@@ -74,6 +74,10 @@ describe('ProtocolOverflowMenu', () => {
     fireEvent.click(button)
     const runButton = getByText('Run now')
     fireEvent.click(runButton)
+    expect(mockTrackEvent).toHaveBeenCalledWith({
+      name: 'proceedToRun',
+      properties: { sourceLocation: 'ProtocolLanding' },
```
Comment (koji) [inline app/src/organisms/ProtocolsLanding/__tests__/ProtocolOverflowMenu.test.tsx]:
```suggestion
      properties: { sourceLocation: 'ProtocolsLanding' },
```

File: app/src/organisms/ProtocolsLanding/ProtocolOverflowMenu.tsx [hunk 3]
```diff
@@ -64,6 +64,10 @@ export function ProtocolOverflowMenu(
   const handleClickRun: React.MouseEventHandler<HTMLButtonElement> = e => {
     e.preventDefault()
     e.stopPropagation()
+    trackEvent({
+      name: 'proceedToRun',
+      properties: { sourceLocation: 'ProtocolLanding' },
```
Comment (koji) [inline app/src/organisms/ProtocolsLanding/ProtocolOverflowMenu.tsx]: Mixpanel data will be analyzed by the PM team so I think using the same name as well as the page would be easy for them.

```suggestion
      properties: { sourceLocation: 'ProtocolsLanding' },
```

Comment (ewagoner) [reply app/src/organisms/ProtocolsLanding/ProtocolOverflowMenu.tsx]: D'oh! It escaped my notice that these were plural. Fixed all four.

**Review by koji — APPROVED** [review]
The changes look good to me!
🤖

**Pull request merged**